### ISSUE TITLE
Doc states no iter_mut on HashMap but there is

### ISSUE THREAD

**Issue by BarePotato** [rust-lang/rust#94755]
https://github.com/rust-lang/rust/blob/1eb72580d076935a3e590deb6e5813a5aef3eca4/library/core/src/iter/mod.rs#L246

> While many collections offer iter(), not all offer iter_mut(). For example, mutating the keys of a [HashSet<T>](https://doc.rust-lang.org/std/collections/struct.HashSet.html) or [HashMap<K, V>](https://doc.rust-lang.org/std/collections/struct.HashMap.html) could put the collection into an inconsistent state if the key hashes change, so these collections only offer iter().

HashMap clearly has iter_mut https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.iter_mut while HashSet does not.

Docs should be changed to reflect this. I would be happy to do this, but as I have not contributed to Rust before, wanted to make sure I didn't miss something here.

**Comment by BarePotato** [comment]
I also realize that the key itself is reference as seen here https://doc.rust-lang.org/std/collections/hash_map/struct.IterMut.html#associatedtype.Item but feel the wording here being updated is still valid.

**Comment by compiler-errors** [comment]
I agree this wording should be updated. Perhaps "or `HashMap<K, V>`" should just be removed.

**Comment by compiler-errors** [comment]
@BarePotato are you still planning on making this change?
